MarketplaceServiceTests/Repository/MarketplaceRepositoryTests.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System;
+using marketplaceservice.DatastoreSettings;
+using marketplaceservice.Domain;
+using marketplaceservice.Repositories;
+using Mongo2Go;
+using Xunit;
+
+namespace marketplaceservicetests.Repository
+{
+    // : IDisposable
+    public class MarketplaceRepositoryTests : IDisposable
+    {
+        public MarketplaceRepositoryTests()
+        {
+            _mongoDbRunner = MongoDbRunner.Start();
+            var settings = new MarketplaceDatabaseSettings
+            {
+                ConnectionString = _mongoDbRunner.ConnectionString,
+                DatabaseName = "IntegrationTests",
+                MarketplaceCollectionName = "TestCollection"
+            };
+            _marketplaceRepository = new MarketplaceRepository(settings);
+        }
+
+        public void Dispose()
+        {
+            _mongoDbRunner.Dispose();
+        }
+
+        private readonly MongoDbRunner _mongoDbRunner;
+        private readonly IMarketplaceRepository _marketplaceRepository;
+
+        [Fact]
+        public async void CreateProduct()
```

Comment:
Async kan niet een void returnen. Maak ipv van void hier overal Task van.

https://docs.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming